From 7e341687ee6de56fcff14111f2f578283060d306 Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Thu, 13 Feb 2020 08:31:13 -0700
Subject: [PATCH 1197/1851] mtd: spi-nor: Fix compilation warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixed the following kernel compilation warnings

drivers/mtd/spi-nor/spi-nor.c:5268:11: warning: return discards ‘const’
qualifier from pointer target type [-Wdiscarded-qualifiers]
    return jinfo;
           ^~~~~
drivers/mtd/spi-nor/spi-nor.c:5283:9: warning: return discards ‘const’
qualifier from pointer target type [-Wdiscarded-qualifiers]
  return info;
         ^~~~~

Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index b87f94f54165..43ce6e37ed3a 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -5265,7 +5265,7 @@ static struct flash_info *spi_nor_get_flash_info(struct spi_nor *nor,
 
 		jinfo = spi_nor_read_id(nor);
 		if (IS_ERR(jinfo)) {
-			return jinfo;
+			return (struct flash_info *)jinfo;
 		} else if (jinfo != info) {
 			/*
 			 * JEDEC knows better, so overwrite platform ID. We
@@ -5280,7 +5280,7 @@ static struct flash_info *spi_nor_get_flash_info(struct spi_nor *nor,
 		}
 	}
 
-	return info;
+	return (struct flash_info *)info;
 }
 
 int spi_nor_scan(struct spi_nor *nor, const char *name,
-- 
2.31.1

